src/@batch-flask/ui/duration-picker/duration-picker.component.ts

```diff
@@ -246,6 +250,18 @@ export class DurationPickerComponent implements FormFieldControl<any>,
                 if (isNaN(time) || time < 0) {
                     this.invalidTimeNumber = true;
                     return null;
+                } else if (time > Constants.MaxDurations.maxDays && this.unit === "days") {
+                    this.invalidDurationValue = true;
+                    return null;
+                } else if (time > Constants.MaxDurations.maxHours && this.unit === "hours") {
+                    this.invalidDurationValue = true;
+                    return null;
+                } else if (time > Constants.MaxDurations.maxMinutes && this.unit === "minutes") {
+                    this.invalidDurationValue = true;
+                    return null;
```

Comment:
move null outside switch statement now that its being reused a bunch

Reply:
Shouldn't this `return null` remain inside the default case?

Reply:
Basically something like this
```
        switch (this.unit) {
            case DurationUnit.Custom:
                return this._getCustomDuration(this.time);
            default:
                const time = Number(this.time);
                if (isNaN(time) || time < 0) {
                    this.invalidTimeNumber = true;
                } else {
                    return Duration.fromObject({
                        [this.unit]: Number(this.time),
                    });
                }
        }
        return null;
```